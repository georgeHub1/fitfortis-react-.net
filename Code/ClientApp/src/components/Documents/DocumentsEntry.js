import React, { Component } from 'react';
import { Breadcrumb } from 'antd';
import styles from './styles.module.less';
import { Link } from 'react-router-dom';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';
export class DocumentsEntry extends Component {
  displayName = DocumentsEntry.name

  componentDidMount () {
    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.onUpdateLanguage(lan);
      localStorage.setItem('language', lan);
    }
    analyticId.firebaseAnalyticsLog('Documents_ViewDocument');
  }
  render () {
    const { location } = this.props;

    const pathSnippets = location
      ? location.pathname.split('/').filter(i => i)
      : [];

    const breadcrumbItems = pathSnippets.map((_, index) => {
      const url = `/${pathSnippets.slice(0, index + 1).join('/')}`;

      return (
        <Breadcrumb.Item key={url}>
          <Link to={url} className={styles.navigationBreadCumpLink}>{_}</Link>
        </Breadcrumb.Item>
      );
    });

    return (
      <div>
        <h1 className={`${styles.header} ${styles.mainHeader}`}>
          {pathSnippets[pathSnippets.length - 1]}
        </h1>
        <div className={styles.navigation}>
          <Breadcrumb className={styles.navigationBreadCump}>
            {breadcrumbItems}
          </Breadcrumb>
        </div>
      </div>
    );
  }
}
