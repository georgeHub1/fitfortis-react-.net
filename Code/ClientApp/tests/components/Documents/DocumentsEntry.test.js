import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter  } from 'react-router-dom';
import { DocumentsEntry } from '../../../src/components/Documents/DocumentsEntry.js';

const Container = DocumentsEntry;

describe('DocumentsEntry', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Container/>)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
