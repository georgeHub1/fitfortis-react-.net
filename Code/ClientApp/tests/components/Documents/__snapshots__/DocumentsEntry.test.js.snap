// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsEntry renders correctly 1`] = `
<div>
  <h1
    className="undefined undefined"
  />
  <div>
    <div
      className="ant-breadcrumb"
    />
  </div>
</div>
`;
